hide the pagination if grid rows are less

swetha571 opened the pull request:
# Type of change? <!-- [x] all the boxes that apply -->

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
- Hide the pagination when Number of rows is less as per the requirement. Devices, rules page grids should have 50 rows per page, if the rows are less than 50 per page then we don't show the pagination.
- AlarmGrid should have 5 rows per systemalarm panel. If they exceed more than 5 then we show pagination.

...

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

dluc commented on src/components/alarmList/alarmList.js:
```diff
@@ -133,7 +133,8 @@ class AlarmList extends Component {
         <div className="grid-container">
-          <AlarmsGrid {...alarmsGridProps} />
+          {this.state.rowData &&
+          <AlarmsGrid {...alarmsGridProps} pagination = {this.state.rowData.length > 8 ? true: false}/>}
```
is "8" the right value in all the situations? if I remember correctly the number of alarms in a page depends on the browser window size, the zoom settings etc.

dluc commented on src/components/alarmList/alarmsGridConfig.js:
```diff
@@ -8,7 +8,7 @@ import { gridValueFormatters } from '../pcsGrid/pcsGridConfig';
 const { checkForEmpty } = gridValueFormatters;
 
-export const DEFAULT_ALARM_GRID_PAGE_SIZE = 8;
+export const DEFAULT_ALARM_GRID_PAGE_SIZE = 5;
```
in `alarmList.js` the value is 8, here it's 5. Are they supposed to be different or the same?   Also, how did you calculate the value before (8) and now (5) ?

swetha571 replied:
It's 5. Because the rows goes beyond the alarmGrid panel if it exceeds 7.

dluc commented on src/components/pages/devices/devices.js:
```diff
@@ -86,7 +86,8 @@ class DevicesPage extends Component {
           </div>
-          <DevicesGrid {...deviceGridProps} />
+          {devices &&
+          <DevicesGrid {...deviceGridProps} pagination={(devices|| []).length > 50 ? true :  false}/>}
```
what is "50" in this context and how was the value calculated? could you use a constant instead of hardcoding the value?

swetha571 replied:
This value is given by Caitlin. Sure i will use the const instead of hard coding.

swetha571 replied:
This value is given by Caitlin. Sure instead of hard coded value i will add const.

dluc commented on src/components/rulesActionsList/ruleAndActionsConfig.js:
```diff
@@ -9,6 +9,8 @@ import lang from "../../common/lang";
 
 export const LastTriggerDefaultWidth = 310;
 
+export const DEFAULT_RULES_GRID_PAGE_SIZE = 50;
```
why is this constant all upper case, while the previous one is PascalCased? we should fix the inconsistencies.  Also please add a comment describing how the value (50 in this case) is calculated, otherwise it will be pretty much impossible to maintain it

swetha571 replied:
Sure. I will fix the casing issue but the 50 rows per page value is given by Caitlin.

dluc replied:
could you ask Caitlin how the constants were calculated?

also, this "50" is it the same of Config.DEVICES_RULESGRID_ROW, and could we reuse the Config value?

dluc commented on src/components/rulesActionsList/rulesActionsList.js:
```diff
@@ -54,9 +54,11 @@ class RulesActionsList extends Component {
       <div className="rules-actions-container">
-        <PcsGrid {...gridProps} />
+        {(this.props.rowData || []).length > 0 &&
+        <PcsGrid {...gridProps} pagination= {(this.props.rowData || []).length > 50 ? true : false}/>}
```
ditto

swetha571 commented:
Updated the PR with
- Removed hard coded value instead used const values.
- Fixed the inconsistencies const casing issues.
- Caitlin provided the Number of rows per page. I will find out from her about that.

spryor commented on src/components/rulesActionsList/rulesActionsList.js:
```diff
@@ -54,9 +55,11 @@ class RulesActionsList extends Component {
     return (
       <div className="rules-actions-container">
-        <PcsGrid {...gridProps} />
+        {(this.props.rowData || []).length > 0 &&
```
```
{
   ... multi line code
}
```

spryor commented on src/components/rulesActionsList/rulesActionsList.js:
```diff
@@ -54,9 +55,11 @@ class RulesActionsList extends Component {
       <div className="rules-actions-container">
-        <PcsGrid {...gridProps} />
+        {(this.props.rowData || []).length > 0 &&
+        <PcsGrid {...gridProps} pagination= {(this.props.rowData || []).length > Config.DEVICES_RULESGRID_ROWS ? true : false}/>}
```
There is a space after `pagination=`
